Ship_Game/Data/ResourceManager.cs

```diff
@@ -2004,11 +2005,22 @@ static void LoadSunZoneData()
             ZoneDistribution[SunZone.VeryFar] = SunZoneData.CreateDistribution(zones, SunZone.VeryFar);
         }
 
-        static readonly Map<BuildRatio, int[]> BuildRatios = new Map<BuildRatio, int[]>();
+        static readonly Map<BuildRatio, Range[]> BuildRatios = new Map<BuildRatio, Range[]>();
+        static readonly Random random = new Random();
 
-        public static int[] GetFleetRatios(BuildRatio canBuild)
+        public static int[] GetFleetRatios(BuildRatio canBuild, bool IsRandomizedAIFleetSizes)
         {
-            return BuildRatios[canBuild];
+            int[] fleetSize = new int[BuildRatios[canBuild].Length];
+            SeededRandom seededRandom = new SeededRandom();
+
+            for (int i = 0; i < BuildRatios[canBuild].Length; i++)
+            {
+                fleetSize[i] = IsRandomizedAIFleetSizes 
+                    ? (int) Math.Floor(BuildRatios[canBuild][i].Generate(seededRandom))
+                    : (int) BuildRatios[canBuild][i].Min;
+            }

```

Comment:
actually you dont need this loop and an if condition for each member.
return a Range[] and deal with the flag in `SetCounts`